README.md

```diff
@@ -64,3 +64,33 @@ production restrictions and how releases are prepared, etc.
 
 Here you should write what are all of the configurations a user can enter
 when using the project, and which file each config is set if applicable.
+
+## Linters and code formatters
+We are using a series of tools to prettify and lint the code we write:
+
+- Husky: to install and run the pre-commit hooks
+- PHPStan: to do static analysis check on PHP code
+- PHP-CS-Fixer: to remove unused dependencies and do some basic formatting
+- Prettier: to fully format the code
+- Blast: if installed, we run it to test if it's compilable
+- Eslint: TODO: not done yet, we need a FEE to help doing this the right way
+- Git conflict markers: the pre-commit checker tool also checks if the developer didn't stage any Git conflicted files by looking for conflict markers on the staged files.
+ 
+These tools are executed automatically on every commit, only on staged files (except for Blast), and for it to work you need to make sure you executed. Composer and NPM are responsible for making sure husky is installed. And these commands are also available if a developer needs to run the commands manually: 
+
+Global commands:
+
+- composer lint
+- composer format
+
+Specific commands:
+
+- composer phpstan
+- composer eslint (TODO)
+- composer prettier
+- composer blast
+- composer php-cs-fixer
+
+Commands execution also generates a log file with the result at .shell/logs/<tool-name>.log, you can check for PHPStan errors on this file, for example.
+
+**TODO: do we need to have the same commands on NPM?** 
```

Comment:
I believe `eslint` being a front-end part maybe needs to be invoked by `npm`, rendering `composer eslint` redundant.

I would expect `composer` invocations be more about PHP/Blade while `npm` be more about JS/CSS, so `eslint` to be invoked by `npm` sounds good.

Thoughts @m4n1ok ?

Reply:
I believe the `composer lint` or `npm run lint` could just lint everything (phpstan, eslint, blast...) and then we also have commands to lint things separately like `composer phpstan`. I don't mind having the same commands on NPM, like `npm lint` and `npm eslint`, but I prefer to use a single tool than to keep remembering which one I should use for lint this, or that, and if composer can do these things for me, I choose to `composer eslint` over `npm run eslint`, but only because of muscle memory...

Reply:
Re: 1. I'm coming from PHP com in theory I should use composer, but I also have to touch JS from time to time, so yes, it's my muscle memory preferring, but it can also be someone's else. What's the harm of having all scripts available on both tools for all audiences?

Re: 2. Should I remove blast from this? Because Blast is working as a linter here too, to lint Blade view components. Maybe it's not the right tool to lint Blade but this is what we have on a project with @ifox, as we saw that not compiling Blast before committing work was generating errors on deployed Storybook applications.

Re: 3. They have independent options already: 


``` php
        "lint": [
            "sh .shell/linters.sh lint"
        ],
        "format": [
            "sh .shell/linters.sh format"
        ],
        "prettify": [
            "sh .shell/linters.sh prettify"
        ],
        "prettier": [
            "sh .shell/linters.sh prettier"
        ],
        "php-cs-fixer": [
            "sh .shell/linters.sh php-cs-fixer"
        ],
        "phpstan": [
            "sh .shell/linters.sh phpstan"
        ],
        "blast": [
            "sh .shell/linters.sh blast"
        ]
```

Reply:
Moved it to a `/tools` folder, but kept the name `linters.sh`.